### ISSUE TITLE
travis builds "randomly" failing

### ISSUE THREAD

**Issue by ebfe** [voidlinux/void-packages#4909]
See https://travis-ci.org/voidlinux/void-packages/jobs/165454444 for an example.

```
=> schismtracker-20160913_1: running pre_configure ...
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
    LANGUAGE = (unset),
    LC_ALL = (unset),
    LC_COLLATE = "C",
    LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
    LANGUAGE = (unset),
    LC_ALL = (unset),
    LC_COLLATE = "C",
    LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
    LANGUAGE = (unset),
    LC_ALL = (unset),
    LC_COLLATE = "C",
    LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
sh: git: command not found
/usr/bin/m4: stack overflow
```

I think @pullmoll tracked this down to proot? somewhere but i can't find the issue atm.

**Comment by pullmoll** [comment]
It happens with `proot` **and** with `uchroot`, not with `uunshare`.
With the former two the reconfigure of the glibc-locales segfaults.

**Comment by ackalker** [comment]
Perhaps related, `proot` segfaults for me when I try and `proot` into a mounted x86_64-musl installation from  x86_64 host:
```sh
$ proot -S mnt
proot info: pid 4251: terminated with signal 11
$ gdb -q proot
Reading symbols from proot...Reading symbols from /usr/lib/debug//usr/bin/proot...done.
done.
(gdb) run -S mnt
Starting program: /usr/bin/proot -S mnt
proot info: pid 4526: terminated with signal 11
[Inferior 1 (process 4522) exited with code 0377]
(gdb) bt
No stack.
```
Huh? Oh... could be that it is actually a child process spawned by `proot` which is segfaulting. Let's try again:
```sh
$ gdb -q proot
Reading symbols from proot...Reading symbols from /usr/lib/debug//usr/bin/proot...done.
done.
(gdb) set follow-fork-mode child
(gdb) run -S mnt
Starting program: /usr/bin/proot -S mnt
[New process 4406]
proot error: ptrace(TRACEME): Operation not permitted
proot error: execve("/usr/bin/sh"): Operation not permitted
proot info: It seems your kernel contains this bug: https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1202161
To workaround it, set the env. variable PROOT_NO_SECCOMP to 1.
fatal error: see `proot --help`.
[Inferior 2 (process 4406) exited with code 01]
proot error: can't chmod '/tmp/proot-4402-tHGXCB': No such file or directory
```
Hmm, looks like `gdb` might be interfering with `proot`'s inner workings, but let's try the suggested workaround anyway:
```
$ PROOT_NO_SECCOMP=1 proot -S mnt
# _
```
Yup! That appears to work. Hopefully this helps with tracking down the root cause of the problem...if mine is actually related at all to the issue at hand.

**Comment by ackalker** [comment]
Guess I celebrated too soon, there appears to be something very strange going on with mountpoints while inside a `proot` session using the workaround, this certainly needs to be investigated.

**Comment by ebfe** [comment]
Seems fixed with #5186
